modules/doc/content/newsletter/2019_12.md

```diff
@@ -1 +1,32 @@
 # MOOSE Newsletter (December 2019)
+
+## Improved Stochastic Tools Sampler API
+
+The Sampler base class was updated to include a more efficient API for computing sample data in
+parallel with minimal memory impact. Please refer to [distributed_samples.md] for more information.
+
+## Miscellaneous enhancements and bug fixes
+
+- The TestHarness now properly detects failures in the Part1 test when using the --recover flag.
+- "Timed Print" formatting fixed when a nested call to this utility is made.
+- Don't allow the use of "contains_complete_history" parameter in `FeatureVolumeVectorPostprocessor`.
+- BoundaryRenameGenerator bug fix when `old_boundary_name` is used.
+
+## Python 3 Update
+
+All python-based utilities within MOOSE (e.g., the test harness, peacock, and MooseDocs) have
+been upgraded to utilize Python 3, which some maintaining support for Python 2.
+[Official language support for Python 2](https://www.python.org/doc/sunset-python-2/)
+officially ends on 1/1/2020. Python 2 support for tools in MOOSE will also end at this time.
+To upgrade to use Python 3, please update your MOOSE environment package by following the
+[getting_started/index.md] instructions.
+
+## Solid Mechanics module retirement schedule
+
+The solid mechanics capabilities in the MOOSE modules were originally provided by the `solid_mechanics` module. Because of limitations with the architecture of the material models in that module, a decision was made several years to re-implement the mechanics capabilities in a new module called `tensor_mechanics`. After a significant development effort, we have migrated all of the models in the INL-developed applications that use mechanics capabilities to be based on `tensor_mechanics`, and we are getting ready to remove the `solid_mechanics` module. Use of any models based on `solid_mechanics` in those applications now generates a deprecation warning message, and we will remove the models based on `solid_mechanics` in the INL-developed MOOSE-based applications on June 30th, 2020. We will remove the `solid_mechanics` module from MOOSE one month later, on July 31, 2020.
+
+We have been publicizing that this change has been coming for a long time, and expect that any new models that are being set up now are based on the `tensor_mechanics` module. The BISON team has put together an [extensive guide](https://mooseframework.org/bison/tutorials/mechanics_conversion/conversion_guidelines.html) on how to migrate existing models to use `tensor_mechanics` as part of BISON's code documentation. While there are details in that guide that apply only to BISON models, the general principles are applicable to converting any model that uses mechanics.
```

Comment:
@sapitts The version that is publicly available is much less complete than the current version (although still better than nothing). Any thoughts on how to handle that?  I'm not sure what the schedule is for updating the website.  Maybe we could move general material somewhere on the MOOSE documentation side?

Reply:
@gardnerru says there is no schedule and that it used to be that "we were lucky to update it once a year"; the current version is stuck in export control. Perhaps we should wait on putting this announcement about deprecating solid mechanics until January.

We certainly could move the general information and probably add in the few additional classes....this is a depressing unleased project

Reply:
Are you guys happy with this? Should we get this merged?

Reply:
The more updated migration guide in BISON just passed the review process minutes ago, so we can probably merge this section as written